### PR TITLE
[Data Logging] "ServerConfig" to Logger

### PR DESCRIPTION
**Pull request by dbogunowicz**
Specifies `build_logger` functions that maps from `server_config: ServerConfig` to `deepsparse_logger: BaseLogger`.
Assumes server configuration in the following form:
```yaml
loggers:
    - some_logger_1
    - some_logger_2
       key1: value1
       key2: value2
endpoints:
    - task: question_answering
      route: /unpruned/predict
      model: zoo:nlp/question_answering/bert-base/pytorch/huggingface/squad/base-none
      batch_size: 1
      data_logging:
        - target: pipeline_outputs
          mappings:
           - func: builtins:some_function1
             frequency: 2
           - func: .../user_defined_metric_functions.py:some_function2
             frequency: 3
        - target: pipeline_outputs
           mappings:
              ...
    - task: question_answering
      route: /unpruned/predict
      model: zoo:nlp/question_answering/bert-base/pytorch/huggingface/squad/base-none
      batch_size: 1
      name: question_answering_
      data_logging:
        - target: engine_outputs
          mappings:
           - func: np.mean
             frequency: 4
```

The envisioned hierarchy of loggers in the context of `DeepSparse Server`:
image

### CODE REVIEW

File: src/deepsparse/server/build_logger.py
Comment (corey-nm) [reply src/deepsparse/server/build_logger.py:66]: @bfineran @dbogunowicz i think splitting out filtering behavior from function logger will make parsing logic much simpler (see code snippet above). without splitting there'd have to be some weird combination of multi logger with existing function logger (i'm struggling to even sketch it out). additionally, splitting into MatchLogger allows us to easily set different target loggers for different functions

File: src/deepsparse/server/config.py
Comment (bfineran) [reply src/deepsparse/server/config.py:150]: let's hold off on this for now

File: src/deepsparse/server/config.py
Comment (dbogunowicz) [reply src/deepsparse/server/config.py]: Will add default frequency in the other PR.

File: tests/server/test_build_logger.py
Comment (dbogunowicz) [inline tests/server/test_build_logger.py:154]: For now, I am testing it by pickling the class objects and comparing their pickled definitions. Once this PR gets approved, I will write a more appropriate code for comparison between `logger` and `expected logger`

File: src/deepsparse/server/build_logger.py
Comment (dbogunowicz) [inline src/deepsparse/server/build_logger.py:96]: We should be checking for two possible unwanted scenarios:
- having the same endpoint name (self.task or self.name) for multiple endpoints
- having the same metric function being applied to the same target

This is a mental note for myself to check against those scenarios in the next PR (integrating `build_logger` into the Server).

File: src/deepsparse/server/build_logger.py
Comment (corey-nm) [inline src/deepsparse/server/build_logger.py:34]: I think this looks good so far, we'll probably need to construct a different logger per endpoint, which is different than the original design layed out. that'll just involve breaking apart this function you have here, so pretty easy

File: src/deepsparse/server/build_logger.py
Comment (corey-nm) [inline src/deepsparse/server/build_logger.py:56]: recommend handling this in the function that calls build_logger

File: src/deepsparse/server/build_logger.py
Comment (corey-nm) [inline src/deepsparse/server/build_logger.py:95]: old code i assume?

File: src/deepsparse/server/config.py
Comment (corey-nm) [inline src/deepsparse/server/config.py:150]:
```suggestion
    loggers: Optional[Dict[str, Dict[str, Any]]] = Field(
```
what about this?

**Review by corey-nm — CHANGES_REQUESTED** [review]
great start! super close IMO

**Pull request closed without merging**
